libraries/resource_providers/api200/connection_template_provider.rb

```diff
@@ -15,19 +15,34 @@ module OneviewCookbook
   module API200
     # ConnectionTemplate API200 provider
     class ConnectionTemplateProvider < ResourceProvider
-      def load_resource_from_ethernet
-        return unless @context.associated_ethernet_network
+      def load_connection_template
```

Comment:
This `if` is kind of big, I'd suggest one of two things for improvement:
1. Turn this method more generic and just call it, as in:
like:
```ruby
def load_connection_template(resource_type, resource_name)
  return unless resource_name
  @item.data.delete('name')
  @item['uri'] = load_resource(resource_type, resource_name, :connectionTemplateUri)
end
```
then we'd just call `load_connection_template(:EthernetNetwork, @context.associated_ethernet_network)` , then fc, then fcoe, then net instead of calling the method once.

Or, 

2. maybe use a `case` instead of `if`.

I personally kind of like the option 1 better.